### PR TITLE
BestWorstDef

### PR DESCRIPTION
**Pull request by MillsProvosty**

### CODE REVIEW

File: lib/league_statistics.rb [hunk 1]
```diff
@@ -148,6 +148,22 @@ def worst_offense
     team_id_to_name_converter(worst_offense_team_id)
   end
 
+  def best_defense
+    best_defense_team_id = average_goals_by_team.max_by do |team_id, average_goals|
```
Comment (milevy1) [inline lib/league_statistics.rb]: This is goals scored but for defense, we need to do goals "allowed" by team.

File: test/league_statistics_test.rb [hunk 2]
```diff
@@ -44,6 +44,28 @@ def test_total_goals_by_team
     assert_equal expected, @stat_tracker.total_goals_by_team
   end
 
+  def test_total_goals_allowed_by_team
```
Comment (milevy1) [inline test/league_statistics_test.rb:47]: Good job making a test for your helper method.

**Comment by milevy1** [comment]
Approved

**Comment by smainar** [comment]
Good work, Mills!

**Pull request merged**